subprojects/plugins/src/main/java/org/gradle/api/plugins/TestReportAggregationPlugin.java

```diff
@@ -0,0 +1,121 @@
+
+        ObjectFactory objects = project.getObjects();
+
+        // prepare testReportDir with a reasonable default, but override with JavaPluginExtension#testReportDir if available
+        final DirectoryProperty testReportDir = objects.directoryProperty().convention(reporting.getBaseDirectory().dir(TestingBasePlugin.TESTS_DIR_NAME));
+        JavaPluginExtension javaPluginExtension = project.getExtensions().findByType(JavaPluginExtension.class);
+        if (javaPluginExtension != null) {
+            testReportDir.set(javaPluginExtension.getTestReportDir());
+        }
+
+        // iterate and configure each user-specified report, creating a <reportName>ExecutionData configuration for each
+        reporting.getReports().withType(AggregateTestReport.class).configureEach(report -> {
+            report.getReportTask().configure(task -> {
+
+                // A resolvable configuration to collect test results; typically named "testResults"
+                Configuration testResultsConf = project.getConfigurations().create(report.getName() + "Results");
+                testResultsConf.extendsFrom(testAggregation);
+                testResultsConf.setDescription(String.format("Supplies test result data to the %s.  External library dependencies may appear as resolution failures, but this is expected behavior.", report.getName()));
+                testResultsConf.setVisible(false);
+                testResultsConf.setCanBeConsumed(false);
+                testResultsConf.setCanBeResolved(true);
+                testResultsConf.attributes(attributes -> {
+                    attributes.attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category.class, Category.DOCUMENTATION));
+                    attributes.attribute(DocsType.DOCS_TYPE_ATTRIBUTE, objects.named(DocsType.class, DocsType.TEST_RESULTS));
+                    attributes.attributeProvider(TestType.TEST_TYPE_ATTRIBUTE, report.getTestType().map(tt -> objects.named(TestType.class, tt)));
+                    attributes.attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage.class, Usage.VERIFICATION));
+                });
+
+                Callable<FileCollection> testResults = () ->
+                    testResultsConf.getIncoming().artifactView(view -> {
+                        view.componentFilter(id -> id instanceof ProjectComponentIdentifier);
+                        view.lenient(true);
+                    }).getFiles();
+
+                task.getTestResults().from(testResults);
+                task.getDestinationDirectory().convention(testReportDir.dir(report.getTestType().map(tt -> tt + "/aggregated-results")));
+            });
+        });
+
+        // convention for synthesizing reports based on existing test suites in "this" project
+        project.getPlugins().withId("jvm-test-suite", p -> {
+            // Depend on this project for aggregation
+            project.getDependencies().add(TEST_REPORT_AGGREGATION_CONFIGURATION_NAME, project);
```

Comment:
Is this right?  The tests pass, so I guess so.  It looks like the project is adding a dep on itself though, and gettings its own extensions rather than the extensions of the other project.  And the `p` for the closure variable also suggests project, `plugin` might be better.

Reply:
This reacts to the situation where the project doing the aggregation also has `jvm-test-suite` applied. 

This loop will create an aggregation task for each test suite present (`unit-tests`, `integration-tests` etc.).

Without the project dependency on line 108, the aggregation result will be missing "this" project's results.

I can change the lambda parameter though; no problem.